array_test.go

```diff
@@ -133,38 +133,63 @@ func BenchmarkArraySetEmpty(b *testing.B) {
 }
 
 func TestArrayObjectMemUsage(t *testing.T) {
+	vm := New()
+
 	tests := []struct {
 		name        string
 		mu          *MemUsageContext
 		ao          *arrayObject
 		expected    uint64
+		newExpected uint64
 		errExpected error
 	}{
 		{
 			name: "mem below threshold",
-			mu:   NewMemUsageContext(New(), 88, 5000, 50, 50, TestNativeMemUsageChecker{}),
+			mu:   NewMemUsageContext(vm, 88, 5000, 50, 50, TestNativeMemUsageChecker{}),
 			ao: &arrayObject{
 				values: []Value{
-					New()._newString(newStringValue("key"), nil),
+					vm._newString(newStringValue("key"), nil),
 				},
 			},
-			expected:    41,
+			// array overhead + array baseObject
```

Comment:
[q] How did we determine the array overhead is 8 bytes when I run it I get 24 bytes?

Reply:
I just empty arrays of different [types in the playground](https://go.dev/play/p/BZE3v0A71g2). I wasn't looking at that overall expected. just the `array overhead` part where that value seems to be 8.

Reply:
Ah ok! So the `arrayObject` is a struct and could potentially have `values` be `nil`. But if it is not `nil` then you're right there's an overhead of just storing an empty array of 24 bytes! I'll take another look at this and keep you posted.